game/audio.cc

```diff
@@ -118,19 +118,25 @@ Music::Music(Audio::Files const& files, unsigned int sr, bool preview): srate(sr
 		tracks.emplace(tf.first, std::make_unique<Track>(tf.second, sr));
 	}
 	suppressCenterChannel = config["audio/suppress_center_channel"].b();
-	aubio_tempo_set_silence(aubioTempo.get(), -41.0);
-	aubio_tempo_set_threshold(aubioTempo.get(), 1.25);
 }
 
-unsigned Audio::aubio_win_size = 2048;
-unsigned Audio::aubio_hop_size = 1024;
+unsigned Audio::aubio_win_size = 1536;
```

Comment:
It is safe if using fftw-3

Reply:
Within the windows branch we're using fftw-3...

Reply:
```
fftw3:x86-windows: x86-windows/bin/fftw3.dll                                                                            fftw3:x86-windows: x86-windows/bin/fftw3.pdb                                                                            fftw3:x86-windows: x86-windows/bin/fftw3f.dll                                                                           fftw3:x86-windows: x86-windows/bin/fftw3f.pdb                                                                           fftw3:x86-windows: x86-windows/bin/fftw3l.dll                                                                           fftw3:x86-windows: x86-windows/bin/fftw3l.pdb                                                                           fftw3:x86-windows: x86-windows/debug/bin/fftw3.dll                                                                      fftw3:x86-windows: x86-windows/debug/bin/fftw3.pdb                                                                      fftw3:x86-windows: x86-windows/debug/bin/fftw3f.dll                                                                     fftw3:x86-windows: x86-windows/debug/bin/fftw3f.pdb                                                                     fftw3:x86-windows: x86-windows/debug/bin/fftw3l.dll                                                                     fftw3:x86-windows: x86-windows/debug/bin/fftw3l.pdb                                                                     fftw3:x86-windows: x86-windows/debug/lib/fftw3.lib                                                                      fftw3:x86-windows: x86-windows/debug/lib/fftw3f.lib                                                                     fftw3:x86-windows: x86-windows/debug/lib/fftw3l.lib                                                                     fftw3:x86-windows: x86-windows/debug/lib/pkgconfig/fftw.pc                                                              fftw3:x86-windows: x86-windows/debug/lib/pkgconfig/fftwf.pc                                                             fftw3:x86-windows: x86-windows/debug/lib/pkgconfig/fftwl.pc                                                             fftw3:x86-windows: x86-windows/include/fftw3.f                                                                          fftw3:x86-windows: x86-windows/include/fftw3.f03                                                                        fftw3:x86-windows: x86-windows/include/fftw3.h                                                                          fftw3:x86-windows: x86-windows/include/fftw3l.f03                                                                       fftw3:x86-windows: x86-windows/include/fftw3q.f03                                                                       fftw3:x86-windows: x86-windows/lib/fftw3.lib                                                                            fftw3:x86-windows: x86-windows/lib/fftw3f.lib                                                                           fftw3:x86-windows: x86-windows/lib/fftw3l.lib                                                                           fftw3:x86-windows: x86-windows/lib/pkgconfig/fftw.pc                                                                    fftw3:x86-windows: x86-windows/lib/pkgconfig/fftwf.pc                                                                   fftw3:x86-windows: x86-windows/lib/pkgconfig/fftwl.pc                                                                   fftw3:x86-windows: x86-windows/share/fftw3/FFTW3Config.cmake                                                            fftw3:x86-windows: x86-windows/share/fftw3/FFTW3ConfigVersion.cmake                                                     fftw3:x86-windows: x86-windows/share/fftw3/FFTW3LibraryDepends-debug.cmake                                              fftw3:x86-windows: x86-windows/share/fftw3/FFTW3LibraryDepends-release.cmake                                            fftw3:x86-windows: x86-windows/share/fftw3/FFTW3LibraryDepends.cmake                                                    fftw3:x86-windows: x86-windows/share/fftw3/copyright                                                                    fftw3:x86-windows: x86-windows/share/fftw3/vcpkg_abi_info.txt                                                           fftw3:x86-windows: x86-windows/share/fftw3f/FFTW3fConfig.cmake                                                          fftw3:x86-windows: x86-windows/share/fftw3f/FFTW3fConfigVersion.cmake                                                   fftw3:x86-windows: x86-windows/share/fftw3f/FFTW3fLibraryDepends-debug.cmake                                            fftw3:x86-windows: x86-windows/share/fftw3f/FFTW3fLibraryDepends-release.cmake                                          fftw3:x86-windows: x86-windows/share/fftw3f/FFTW3fLibraryDepends.cmake                                                  fftw3:x86-windows: x86-windows/share/fftw3l/FFTW3lConfig.cmake                                                          fftw3:x86-windows: x86-windows/share/fftw3l/FFTW3lConfigVersion.cmake                                                   fftw3:x86-windows: x86-windows/share/fftw3l/FFTW3lLibraryDepends-debug.cmake                                            fftw3:x86-windows: x86-windows/share/fftw3l/FFTW3lLibraryDepends-release.cmake                                          fftw3:x86-windows: x86-windows/share/fftw3l/FFTW3lLibraryDepends.cmake                                                  fftw3:x64-windows: x64-windows/bin/fftw3.dll                                                                            fftw3:x64-windows: x64-windows/bin/fftw3.pdb                                                                            fftw3:x64-windows: x64-windows/bin/fftw3f.dll                                                                           fftw3:x64-windows: x64-windows/bin/fftw3f.pdb                                                                           fftw3:x64-windows: x64-windows/bin/fftw3l.dll                                                                           fftw3:x64-windows: x64-windows/bin/fftw3l.pdb                                                                           fftw3:x64-windows: x64-windows/debug/bin/fftw3.dll                                                                      fftw3:x64-windows: x64-windows/debug/bin/fftw3.pdb                                                                      fftw3:x64-windows: x64-windows/debug/bin/fftw3f.dll                                                                     fftw3:x64-windows: x64-windows/debug/bin/fftw3f.pdb                                                                     fftw3:x64-windows: x64-windows/debug/bin/fftw3l.dll                                                                     fftw3:x64-windows: x64-windows/debug/bin/fftw3l.pdb                                                                     fftw3:x64-windows: x64-windows/debug/lib/fftw3.lib                                                                      fftw3:x64-windows: x64-windows/debug/lib/fftw3f.lib                                                                     fftw3:x64-windows: x64-windows/debug/lib/fftw3l.lib                                                                     fftw3:x64-windows: x64-windows/debug/lib/pkgconfig/fftw.pc                                                              fftw3:x64-windows: x64-windows/debug/lib/pkgconfig/fftwf.pc                                                             fftw3:x64-windows: x64-windows/debug/lib/pkgconfig/fftwl.pc                                                             fftw3:x64-windows: x64-windows/include/fftw3.f                                                                          fftw3:x64-windows: x64-windows/include/fftw3.f03                                                                        fftw3:x64-windows: x64-windows/include/fftw3.h                                                                          fftw3:x64-windows: x64-windows/include/fftw3l.f03                                                                       fftw3:x64-windows: x64-windows/include/fftw3q.f03                                                                       fftw3:x64-windows: x64-windows/lib/fftw3.lib                                                                            fftw3:x64-windows: x64-windows/lib/fftw3f.lib                                                                           fftw3:x64-windows: x64-windows/lib/fftw3l.lib                                                                           fftw3:x64-windows: x64-windows/lib/pkgconfig/fftw.pc                                                                    fftw3:x64-windows: x64-windows/lib/pkgconfig/fftwf.pc                                                                   fftw3:x64-windows: x64-windows/lib/pkgconfig/fftwl.pc                                                                   fftw3:x64-windows: x64-windows/share/fftw3/FFTW3Config.cmake                                                            fftw3:x64-windows: x64-windows/share/fftw3/FFTW3ConfigVersion.cmake                                                     fftw3:x64-windows: x64-windows/share/fftw3/FFTW3LibraryDepends-debug.cmake                                              fftw3:x64-windows: x64-windows/share/fftw3/FFTW3LibraryDepends-release.cmake                                            fftw3:x64-windows: x64-windows/share/fftw3/FFTW3LibraryDepends.cmake                                                    fftw3:x64-windows: x64-windows/share/fftw3/copyright                                                                    fftw3:x64-windows: x64-windows/share/fftw3/vcpkg_abi_info.txt                                                           fftw3:x64-windows: x64-windows/share/fftw3f/FFTW3fConfig.cmake                                                          fftw3:x64-windows: x64-windows/share/fftw3f/FFTW3fConfigVersion.cmake                                                   fftw3:x64-windows: x64-windows/share/fftw3f/FFTW3fLibraryDepends-debug.cmake                                            fftw3:x64-windows: x64-windows/share/fftw3f/FFTW3fLibraryDepends-release.cmake                                          fftw3:x64-windows: x64-windows/share/fftw3f/FFTW3fLibraryDepends.cmake                                                  fftw3:x64-windows: x64-windows/share/fftw3l/FFTW3lConfig.cmake                                                          fftw3:x64-windows: x64-windows/share/fftw3l/FFTW3lConfigVersion.cmake                                                   fftw3:x64-windows: x64-windows/share/fftw3l/FFTW3lLibraryDepends-debug.cmake                                            fftw3:x64-windows: x64-windows/share/fftw3l/FFTW3lLibraryDepends-release.cmake                                          fftw3:x64-windows: x64-windows/share/fftw3l/FFTW3lLibraryDepends.cmake 
```

Reply:
@Lord-Kamina According to the [FFTW-3 docs](http://www.fftw.org/fftw3_doc/Introduction.html)

> For example, the standard FFTW distribution works most efficiently for arrays whose size can be factored into small primes (2, 3, 5, and 7), and otherwise it uses a slower general-purpose routine. 

So i think we should consider using a small prime like 2 as @allcreater is saying.

Keeping `aubio_win_size` 2048 and  `aubio_hop_size` 1024 resolves the problem on our end.

According to the [aubio docs](https://aubio.org/doc/0.4.0/tempo_8h.html#aaa49f5d6554904a452d031775f385521) it indeed returns `NULL` if the call to `new_aubio_tempo` is not succesful.

Reply:
> @Lord-Kamina According to the [FFTW-3 docs](http://www.fftw.org/fftw3_doc/Introduction.html)
> 
> > For example, the standard FFTW distribution works most efficiently for arrays whose size can be factored into small primes (2, 3, 5, and 7), and otherwise it uses a slower general-purpose routine.
> 
> So i think we should consider using a small prime like 2 as @allcreater is saying.
> 
> Keeping `aubio_win_size` 2048 and `aubio_hop_size` 1024 resolves the problem on our end.
> 
> According to the [aubio docs](https://aubio.org/doc/0.4.0/tempo_8h.html#aaa49f5d6554904a452d031775f385521) it indeed returns `NULL` if the call to `new_aubio_tempo` is not succesful.

If that is indeed confirmed to be the issue I'll fix it but I'm not sure that is the issue. From a question I asked upstream a while ago (https://github.com/aubio/aubio/issues/226#issuecomment-454403848):

> hi @Lord-Kamina
>
> you can use non-power of two hop size regardless of the fft implementation. only fftw3 supports non-power-of-two window sizes, and power-of-two sizes should be preferred as much as possible.

In my testing, the npot win-size gave better results. I have to see, however, what the result will be when I add the resampling to 44.1kHz instead of 48kHz for the preview.  Maybe that will negate the difference.

Reply:
Also, 1536 is (2^9 *3) so it definitely still fits the bill of what that documentation says.